pod/perlre.pod

```diff
@@ -1167,18 +1161,88 @@ X<\g{1}> X<\g{-1}> X<\g{name}> X<relative backreference> X<named backreference>
 X<named capture buffer> X<regular expression, named capture buffer>
 X<named capture group> X<regular expression, named capture group>
 X<%+> X<$+{name}> X<< \k<name> >>
-There is no limit to the number of captured substrings that you may use.
-Groups are numbered with the leftmost open parenthesis being number 1, I<etc>.  If
-a group did not match, the associated backreference won't match either. (This
-can happen if the group is optional, or in a different branch of an
-alternation.)
-You can omit the C<"g">, and write C<"\1">, I<etc>, but there are some issues with
-this form, described below.
-
-You can also refer to capture groups relatively, by using a negative number, so
-that C<\g-1> and C<\g{-1}> both refer to the immediately preceding capture
-group, and C<\g-2> and C<\g{-2}> both refer to the group before it.  For
-example:
+
+The grouping construct C<( ... )> creates a capture group (also
+referred to as a "capture buffer" or sometimes simply "capture" or
+"buffer"). There is no limit to the number of capture groups that you
+may use. Capture groups are numbered with the leftmost open parenthesis
+being number 1, the second from the left being 2, I<etc>, with one
+exception, capture groups inside of the "branch reset" construct
+C<(?| ... )> are numbered differently. (See the section on
+L<branch reset|/Branch Reset> later in this file.)
+
+Capture bufers may also be given a name by appending C<< ?<name> >>
+immediately after the open paren, for example C<< (?<blah> ... ) >>
+creates a capture group called C<blah>. This will be covered in more
+detail a bit later on in this section.
+
+After a successful match the contents of a capture group may be accessed
+via a variety of different L<special variables|perlvar/Regex Variables>
+which can be used to access the content of the buffers by number or
+name, as well as to extract the offsets into the string that was
+matched, there is also a special variable that allows access to the
+capture groups as a whole in a single array or the named captures via a
+hash.
+
+The oldest, most common and traditional way to access the contents of a
+capture buffer after a successful match is via a
+L<digit var|perlvar/Digit Vars> such as C<$1> or C<$2> I<etc>, which refer
+to the first and second capture groups in the pattern. You may refer to
+a named capture group by using C<$+{name}>. Multiple capture groups may
+be given the same name, and you may access the contents of all of the
+capture groups with the same name via the array returned by C<$+{name}>.
```

Comment:
> The same name may be used for more than one capture group.  You can use C<$+{name}> ...

but it doesn't make sense to me that an array is accessed via a dollar sigil, unless it is a reference, and you need to say $+{name}->[0] etc

Reply:
Yes it is a reference. I was more thinking of code like

```
my $all_captures_for_foo = $+{foo};
print "@$all_captures_for_foo";
```

but ill figure out a better wording.

Reply:
This doesn't seem accurate? From perlvar: "If multiple distinct capture groups have the same name, then $+{NAME} will refer to the leftmost defined group in the match."

Reply:
> This doesn't seem accurate?

the `$+{NAME}` is a typo, it should say `$-{NAME}` in that example. Good catch.